apiserver/facades/agent/uniter/service.go

```diff
@@ -97,6 +98,14 @@ type ApplicationService interface {
 	// WatchApplication returns a NotifyWatcher for changes to the application.
 	WatchApplication(ctx context.Context, name string) (watcher.NotifyWatcher, error)
 
+	// WatchUnitForLegacyUniter watches for some specific changes to the unit with
+	// the given name. The watcher will emit a notification when there is a change to
+	// the unit's inherent properties, it's subordinates or it's resolved mode.
+	//
+	// If the unit does not exist an error satisfying [applicationerrors.UnitNotFound]
+	// will be returned.
+	WatchUnitForLegacyUniter(ctx context.Context, unitName coreunit.Name) (watcher.NotifyWatcher, error)
```

Comment:
Can we rename this (I know it wasn't added in this patch)? It looks like a temporary method on a new domain, not great.

Reply:
Discussed offline